src/sentry/rules/actions/utils.py

```diff
@@ -62,26 +79,37 @@ def get_frequency_label(value_str: str | None) -> str | None:
     return None
 
 
+def convert_data(data: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    converted_data = defaultdict(list)
+    for datum in data:
+        id = datum["id"]
+        values = {}
+        for d in datum:
+            values[d] = datum[d]
```

Comment:
🤔 what are we doing this for?
are we just trying to clone the dict?

rather than this helper, can you not just do

```
data_by_id = {[d.id]: d for d in data}
```
?

Reply:
The data is a list of dictionaries, so I was converting it to a dictionary where the key is the id and the value is the rest of the data for that id.

Reply:
I cleaned this up, it was weird because I was trying to not have the id in both the key and the value but then I ended up needing it in both and 🙃